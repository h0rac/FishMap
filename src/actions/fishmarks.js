import {SET_FISHMARK_POSITION, UPDATE_FISHMARK_DATA, MOVE_TO_FISHMARK_POSITION,
    DELETE_FISHMARK_POSITION, LOAD_FISHMARK_POSITIONS, UPLOAD_FISHMARK_POSITIONS
} from '../constants/constants'

export const setFishmark = (data) => ({
    type: SET_FISHMARK_POSITION,
    data,
});

export const moveToFishmarkPosition = (position,selected) => ({
    type: MOVE_TO_FISHMARK_POSITION,
    position,
    selected
});

export const deleteFishmarkPosition = (position) => ({
    type: DELETE_FISHMARK_POSITION,
    position,
});

export const loadFishmarkPositions = () => ({
    type: LOAD_FISHMARK_POSITIONS
});

export const uploadFishmarkPositions = data => ({
    type: UPLOAD_FISHMARK_POSITIONS,
    data,
})