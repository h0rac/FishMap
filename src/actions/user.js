import {
	SET_USER_DATA,
	LOGOUT,
	LOGIN,
	VERIFY_TOKEN,
	GET_USER_LOCATION,
	CHANGE_DURATION,
	CHANGE_RECEIVE_STATUS,
	EMIT_WAYPOINT_RECEIVE,
	SET_INTERVAL_ID,
	SET_SELECTED_DURATION,
	SET_IO_SOCKET,
	SET_INTERVAL_ALIVE,
	CREATE_ACCOUNT,
	RESEND_VERIFICATION_EMAIL
} from '../constants/constants';

export const setUserData = (data) => ({
	type: SET_USER_DATA,
	data
});

export const logout = () => ({
	type: LOGOUT
});

export const login = (data) => ({
	type: LOGIN,
	data
});

export const createAccount = (data) => ({
	type: CREATE_ACCOUNT,
	data,

});

export const resendVerifyEmail = (data) => ({
	type: RESEND_VERIFICATION_EMAIL,
	data
})

export const checkAuthToken = (navigate, screen) => ({
	type: VERIFY_TOKEN,
	navigate,
	screen
});

export const getUserLocation = () => ({
	type: GET_USER_LOCATION
});

export const changeReceiveStatus = (status) => ({
	type: CHANGE_RECEIVE_STATUS,
	receive: status

});

export const changeDuration = (tempDuration) => ({
	type: CHANGE_DURATION,
	tempDuration: tempDuration

});

export const emitWaypointReceive = (emitStatus) => ({
	type: EMIT_WAYPOINT_RECEIVE,
	emitStatus: emitStatus

});

export const setIntervalID = (timeoutID) => ({
	type: SET_INTERVAL_ID,
	timeoutID: timeoutID
});

export const setSelectedDuration = (duration) => ({
	type: SET_SELECTED_DURATION,
	duration: duration
});

export const setIOSocket = (socket) => ({
	type: SET_IO_SOCKET,
	socket: socket
});

export const setIntervalAlive = (status) => ({
	type: SET_INTERVAL_ALIVE,
	status: status
});
