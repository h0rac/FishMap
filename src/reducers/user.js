import {
  SET_USER_DATA,
  SUCCESS_SET_TOKEN,
  FAILED_SET_TOKEN,
  FAILED_GET_USER_LOCATION,
  SUCCESS_GET_USER_LOCATION,
  SUCCESS_VERIFY_TOKEN,
  CHANGE_RECEIVE_STATUS,
  EMIT_WAYPOINT_RECEIVE_STARTED,
  EMIT_WAYPOINT_RECEIVE_STOPPED,
  SET_INTERVAL_ID,
  SET_SELECTED_DURATION, CHANGE_DURATION_SUCCESS,
  SET_IO_SOCKET,
  SET_INTERVAL_ALIVE,
  SET_DST_EMAIL,

}
from '../constants/constants';

const intialState = {
  user: {},
  success: false,
  error: false,
  isFetching: true,
  position: {},
  message: null,
  duration: 8000,
  tempDuration: 0,
  receive: true,
  emitStatus: false,
  timeoutID: 0,
  intervalAlive: true,
  dstEmail: null,
  socketIO: null,
};

const reducer = (state = intialState, action) => {
switch (action.type) {
  case SET_USER_DATA:
    return {
      ...state, user: action.data,
    };

case SUCCESS_SET_TOKEN:
  return {
    ...state, success: action.success, socketIO: action.socketIO,
  };
case FAILED_SET_TOKEN:
  return {
    ...state, error: action.error,
  };

case SUCCESS_GET_USER_LOCATION:
  return {
    ...state, position: action.position, isFetching: action.isFetching,
  };

case FAILED_GET_USER_LOCATION:
  return {
    ...state, error: action.error,
  };
case SUCCESS_VERIFY_TOKEN:
  return {
    ...state,
  };

case CHANGE_RECEIVE_STATUS:
return {
  ...state, receive: action.receive,
};

case CHANGE_DURATION_SUCCESS:
  return {
    ...state, duration: action.duration,
  };

case EMIT_WAYPOINT_RECEIVE_STOPPED:
  return {
    ...state, emitStatus: action.emitStatus, intervalAlive: action.intervalAlive,
  };

case EMIT_WAYPOINT_RECEIVE_STARTED:
  return {
    ...state, emitStatus: action.emitStatus, intervalAlive: action.intervalAlive,
  };

case SET_INTERVAL_ID:
  return {
    ...state, timeoutID: action.timeoutID,
  };

case SET_SELECTED_DURATION:
  return {
    ...state, tempDuration: action.duration,
  };

case SET_IO_SOCKET:
  return {
    ...state, socketIO: action.socket,
  };

case SET_INTERVAL_ALIVE:
  return {
    ...state, intervalAlive: action.status,
  };

case SET_DST_EMAIL:
  return {
    ...state, dstEmail: action.email,
  };

default:
  return state;
 }
};

export default reducer;
