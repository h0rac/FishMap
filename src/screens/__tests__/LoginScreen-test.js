import { TextInput, Keyboard, Dimensions, Platform } from 'react-native';
import React from 'react';
import LoginScreen from '../LoginScreen';
import TestUtils from 'react-dom/test-utils';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import { checkAuthToken } from '../../actions/user';

configure({ adapter: new Adapter() });

const middlewares = [];
const mockStore = configureStore(middlewares);

jest.mock('Platform', () => {
		return {
			Platform: { OS: 'ios' }
		}
	}
)

describe('Testing Login screen', () => {

	let tree;
	let navigate;
	let state;
	let props;
	let dispatch;

	beforeEach(() => {
		navigate = jest.fn();
		state = {
			window: { screen: { height: 375, width: 667 } }
		};
		dispatch = jest.fn();
		props = {
			dispatch
		};
		tree = shallow(
			<LoginScreen {...props} store={mockStore()} navigation={{ navigate }} state={state}
			/>);
	});

	it('renders correctly', () => {
		expect(tree.dive()).toMatchSnapshot();
	});

	it('should show keyboard on listener event', () => {
		const LoginScreenComponent = tree.dive();
		LoginScreenComponent.setState({ keyboardShow: false });
		const keyboardDidShow = jest.spyOn(LoginScreenComponent.instance(), '_keyboardDidShow');
		keyboardDidShow();
		expect(LoginScreenComponent.state().keyboardShow).toBe(true);
	});

	it('should hide keyboard on listener event', () => {
		const LoginScreenComponent = tree.dive();
		LoginScreenComponent.setState({ keyboardShow: true });
		const keyboardDidHide = jest.spyOn(LoginScreenComponent.instance(), '_keyboardDidHide');
		keyboardDidHide();
		expect(LoginScreenComponent.state().keyboardShow).toBe(false);
	});

	it('should invoke dimension handler', () => {
		const LoginScreenComponent = tree.dive();
		const dimensionHandler = jest.spyOn(LoginScreenComponent.instance(), '_DimensionHandler');
		const window = { screen: { height: 375, width: 667 } };
		dimensionHandler(window);
	});

	it('should remove all listeners on component unmount', () => {
		const LoginScreenComponent = tree.dive();
		const componentWillUnmount = jest.spyOn(LoginScreenComponent.instance(), 'componentWillUnmount');
		componentWillUnmount();
		expect(componentWillUnmount).toHaveBeenCalled();
	});

	it('should work in portrait mode', () => {
		const LoginScreenComponent = tree.dive();
		LoginScreenComponent.setState({ window: { screen: { height: 667, width: 375 } } });
		const checkScreenOrientation = jest.spyOn(LoginScreenComponent.instance(), 'checkScreenOrientation');
		const mode = checkScreenOrientation();
		expect(mode).toEqual('portrait');
	});

	it('should work in landscape mode', () => {
		const LoginScreenComponent = tree.dive();
		LoginScreenComponent.setState({ window: { screen: { height: 375, width: 667 } } });
		const checkScreenOrientation = jest.spyOn(LoginScreenComponent.instance(), 'checkScreenOrientation');
		const mode = checkScreenOrientation();
		expect(mode).toEqual('landscape');
	});

	it('should dispatch verification of authentication token', () => {
		dispatch(checkAuthToken(navigate));
		expect(dispatch).toHaveBeenCalledWith(checkAuthToken(navigate));

	});


	it('should render 2 TextInputs', () => {
		const LoginScreenComponent = tree.dive();
		const inputs = LoginScreenComponent.find('TextInput');
		expect(inputs.length).toBe(2);
	});

	it('should render 2 IconButtons', () => {
		const LoginScreenComponent = tree.dive();
		const buttons = LoginScreenComponent.find('IconButton');
		expect(buttons.length).toBe(2);
	});

	it('should send API Request when email and password are valid', () => {
		const LoginScreenComponent = tree.dive();
		const buttons = LoginScreenComponent.find('IconButton');
		LoginScreenComponent.setState({ email: 'test@test.com', password: 'pass123' });
		buttons.first().simulate('Press');
	});

	it('should not send API Request when email is invalid', () => {
		const LoginScreenComponent = tree.dive();
		const buttons = LoginScreenComponent.find('IconButton');
		LoginScreenComponent.setState({ email: 'test#test.com', password: 'pass123' });
		buttons.first().simulate('Press');
		expect(LoginScreenComponent.state().disableLogin).toBe(true);
	});

	it('should redirect to Create Account screen when create account button is pressed', () => {
		const LoginScreenComponent = tree.dive();
		const buttons = LoginScreenComponent.find('IconButton');
		buttons.forEach(button => {
			if (button.getElement().props.name === 'user-o')
				button.simulate('Press');
		});
	});


	it('input fields should not be empty', () => {
		const LoginScreenComponent = tree.dive();
		const inputs = LoginScreenComponent.find('TextInput');
		inputs.forEach(input => {
			input.simulate('ChangeText', 'test123');
		});
		expect(LoginScreenComponent.state().email).toBe('test123');
		expect(LoginScreenComponent.state().password).toBe('test123');
	});


	it('Login button should be enabled when email and password is provided', () => {
		const LoginScreenComponent = tree.dive();
		const inputs = LoginScreenComponent.find('TextInput');
		LoginScreenComponent.setState({ email: 'test@false.com', password: 'pass123' });
		inputs.forEach(input => {
			expect(input.simulate('Change'));
		});
		expect(LoginScreenComponent.state().disableLogin).toBe(false);
	});

	it('Login button should not be enabled when email and password are not provided', () => {
		const LoginScreenComponent = tree.dive();
		const inputs = LoginScreenComponent.find('TextInput');
		LoginScreenComponent.setState({ email: '', password: '' });
		inputs.forEach(input => {
			expect(input.simulate('Change'));
		});
		expect(LoginScreenComponent.state().disableLogin).toBe(true);
	});


	it('Login button should not be enabled when email is not provided', () => {
		const LoginScreenComponent = tree.dive();
		const inputs = LoginScreenComponent.find('TextInput');
		LoginScreenComponent.setState({ email: '', password: 'pass123' });
		inputs.forEach(input => {
			expect(input.simulate('Change'));
		});
		expect(LoginScreenComponent.state().disableLogin).toBe(true);
	});

	it('Login button should not be enabled when password is not provided', () => {
		const LoginScreenComponent = tree.dive();
		const inputs = LoginScreenComponent.find('TextInput');
		LoginScreenComponent.setState({ email: 'test@false.com', password: '' });
		inputs.forEach(input => {
			expect(input.simulate('Change'));
		});
		expect(LoginScreenComponent.state().disableLogin).toBe(true);
	});

	it('Login button should not be enabled when email is invalid', () => {
		const LoginScreenComponent = tree.dive();
		const inputs = LoginScreenComponent.find('TextInput');
		LoginScreenComponent.setState({ email: 'test@false.c', password: 'pass123' });
		inputs.forEach(input => {
			expect(input.simulate('Change'));
		});
		expect(LoginScreenComponent.state().disableLogin).toBe(true);
	});

});