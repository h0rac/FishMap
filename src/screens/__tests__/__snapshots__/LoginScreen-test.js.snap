// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Login screen renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginScreen
    clearActions={[Function]}
    dispatch={[Function]}
    getActions={[Function]}
    getState={[Function]}
    navigation={
      Object {
        "navigate": [Function],
      }
    }
    replaceReducer={[Function]}
    state={
      Object {
        "email": "test@test.pl",
        "password": null,
        "window": Object {
          "screen": Object {
            "height": 375,
            "width": 667,
          },
        },
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
    subscribe={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "alignItems": "center",
                "flex": 2,
                "flexDirection": "column",
                "justifyContent": "flex-end",
                "paddingTop": 120,
              },
            ]
          }
        >
          <Image
            source={1}
          />
        </View>,
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "backgroundColor": "white",
                "flex": 2,
                "flexDirection": "column",
                "paddingBottom": 40,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 32,
              },
            ]
          }
        >
          <TextInput
            maxLength={40}
            onChangeText={[Function]}
            placeholder="your@email.com"
            placeholderTextColor="lightgray"
            style={
              Object {
                "borderBottomColor": "#2F95D6",
                "borderBottomWidth": 1,
                "color": "gray",
                "height": 50,
              }
            }
            underlineColorAndroid="#2F95D6"
          />
          <TextInput
            maxLength={40}
            onChangeText={[Function]}
            placeholder="password"
            placeholderTextColor="lightgray"
            secureTextEntry={true}
            style={
              Object {
                "borderBottomColor": "#2F95D6",
                "borderBottomWidth": 1,
                "color": "gray",
                "height": 50,
              }
            }
            underlineColorAndroid="#2F95D6"
          />
        </View>,
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "backgroundColor": "white",
                "flex": 3,
                "flexDirection": "column",
                "justifyContent": "space-around",
                "paddingLeft": 20,
                "paddingRight": 20,
              },
            ]
          }
        >
          <Button
            backgroundColor="gray"
            disabled={true}
            icon={
              Object {
                "name": "sign-in",
                "type": "font-awesome",
              }
            }
            onPress={[Function]}
            rounded={true}
            title="Login"
          />
          <View
            style={
              Object {
                "alignItems": "center",
              }
            }
          >
            <TouchableHighlight
              activeOpacity={0.85}
              underlayColor="black"
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
              >
                Forgot password ?
              </Text>
            </TouchableHighlight>
          </View>
        </View>,
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "backgroundColor": "white",
                "flex": 2,
                "justifyContent": "space-around",
                "paddingLeft": 20,
                "paddingRight": 20,
              },
            ]
          }
        >
          <Button
            backgroundColor="green"
            icon={
              Object {
                "name": "user-o",
                "type": "font-awesome",
              }
            }
            onPress={[Function]}
            rounded={true}
            title="Create new Account"
          />
        </View>,
      ],
      "style": Object {
        "backgroundColor": "#2F95D6",
        "flex": 1,
        "justifyContent": "center",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Image
            source={1}
          />,
          "style": Array [
            Object {
              "flex": 1,
            },
            Object {
              "alignItems": "center",
              "flex": 2,
              "flexDirection": "column",
              "justifyContent": "flex-end",
              "paddingTop": 120,
            },
          ],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "source": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <TextInput
              maxLength={40}
              onChangeText={[Function]}
              placeholder="your@email.com"
              placeholderTextColor="lightgray"
              style={
                Object {
                  "borderBottomColor": "#2F95D6",
                  "borderBottomWidth": 1,
                  "color": "gray",
                  "height": 50,
                }
              }
              underlineColorAndroid="#2F95D6"
            />,
            <TextInput
              maxLength={40}
              onChangeText={[Function]}
              placeholder="password"
              placeholderTextColor="lightgray"
              secureTextEntry={true}
              style={
                Object {
                  "borderBottomColor": "#2F95D6",
                  "borderBottomWidth": 1,
                  "color": "gray",
                  "height": 50,
                }
              }
              underlineColorAndroid="#2F95D6"
            />,
          ],
          "style": Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "white",
              "flex": 2,
              "flexDirection": "column",
              "paddingBottom": 40,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 32,
            },
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "maxLength": 40,
              "onChangeText": [Function],
              "placeholder": "your@email.com",
              "placeholderTextColor": "lightgray",
              "style": Object {
                "borderBottomColor": "#2F95D6",
                "borderBottomWidth": 1,
                "color": "gray",
                "height": 50,
              },
              "underlineColorAndroid": "#2F95D6",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "maxLength": 40,
              "onChangeText": [Function],
              "placeholder": "password",
              "placeholderTextColor": "lightgray",
              "secureTextEntry": true,
              "style": Object {
                "borderBottomColor": "#2F95D6",
                "borderBottomWidth": 1,
                "color": "gray",
                "height": 50,
              },
              "underlineColorAndroid": "#2F95D6",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Button
              backgroundColor="gray"
              disabled={true}
              icon={
                Object {
                  "name": "sign-in",
                  "type": "font-awesome",
                }
              }
              onPress={[Function]}
              rounded={true}
              title="Login"
            />,
            <View
              style={
                Object {
                  "alignItems": "center",
                }
              }
            >
              <TouchableHighlight
                activeOpacity={0.85}
                underlayColor="black"
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                >
                  Forgot password ?
                </Text>
              </TouchableHighlight>
            </View>,
          ],
          "style": Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "white",
              "flex": 3,
              "flexDirection": "column",
              "justifyContent": "space-around",
              "paddingLeft": 20,
              "paddingRight": 20,
            },
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "backgroundColor": "gray",
              "disabled": true,
              "icon": Object {
                "name": "sign-in",
                "type": "font-awesome",
              },
              "onPress": [Function],
              "rounded": true,
              "title": "Login",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <TouchableHighlight
                activeOpacity={0.85}
                underlayColor="black"
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                >
                  Forgot password ?
                </Text>
              </TouchableHighlight>,
              "style": Object {
                "alignItems": "center",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "activeOpacity": 0.85,
                "children": <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                >
                  Forgot password ?
                </Text>,
                "underlayColor": "black",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "Forgot password ?",
                  "disabled": false,
                  "ellipsizeMode": "tail",
                },
                "ref": null,
                "rendered": "Forgot password ?",
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Button
            backgroundColor="green"
            icon={
              Object {
                "name": "user-o",
                "type": "font-awesome",
              }
            }
            onPress={[Function]}
            rounded={true}
            title="Create new Account"
          />,
          "style": Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "white",
              "flex": 2,
              "justifyContent": "space-around",
              "paddingLeft": 20,
              "paddingRight": 20,
            },
          ],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "backgroundColor": "green",
            "icon": Object {
              "name": "user-o",
              "type": "font-awesome",
            },
            "onPress": [Function],
            "rounded": true,
            "title": "Create new Account",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                },
                Object {
                  "alignItems": "center",
                  "flex": 2,
                  "flexDirection": "column",
                  "justifyContent": "flex-end",
                  "paddingTop": 120,
                },
              ]
            }
          >
            <Image
              source={1}
            />
          </View>,
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                },
                Object {
                  "backgroundColor": "white",
                  "flex": 2,
                  "flexDirection": "column",
                  "paddingBottom": 40,
                  "paddingLeft": 20,
                  "paddingRight": 20,
                  "paddingTop": 32,
                },
              ]
            }
          >
            <TextInput
              maxLength={40}
              onChangeText={[Function]}
              placeholder="your@email.com"
              placeholderTextColor="lightgray"
              style={
                Object {
                  "borderBottomColor": "#2F95D6",
                  "borderBottomWidth": 1,
                  "color": "gray",
                  "height": 50,
                }
              }
              underlineColorAndroid="#2F95D6"
            />
            <TextInput
              maxLength={40}
              onChangeText={[Function]}
              placeholder="password"
              placeholderTextColor="lightgray"
              secureTextEntry={true}
              style={
                Object {
                  "borderBottomColor": "#2F95D6",
                  "borderBottomWidth": 1,
                  "color": "gray",
                  "height": 50,
                }
              }
              underlineColorAndroid="#2F95D6"
            />
          </View>,
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                },
                Object {
                  "backgroundColor": "white",
                  "flex": 3,
                  "flexDirection": "column",
                  "justifyContent": "space-around",
                  "paddingLeft": 20,
                  "paddingRight": 20,
                },
              ]
            }
          >
            <Button
              backgroundColor="gray"
              disabled={true}
              icon={
                Object {
                  "name": "sign-in",
                  "type": "font-awesome",
                }
              }
              onPress={[Function]}
              rounded={true}
              title="Login"
            />
            <View
              style={
                Object {
                  "alignItems": "center",
                }
              }
            >
              <TouchableHighlight
                activeOpacity={0.85}
                underlayColor="black"
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                >
                  Forgot password ?
                </Text>
              </TouchableHighlight>
            </View>
          </View>,
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                },
                Object {
                  "backgroundColor": "white",
                  "flex": 2,
                  "justifyContent": "space-around",
                  "paddingLeft": 20,
                  "paddingRight": 20,
                },
              ]
            }
          >
            <Button
              backgroundColor="green"
              icon={
                Object {
                  "name": "user-o",
                  "type": "font-awesome",
                }
              }
              onPress={[Function]}
              rounded={true}
              title="Create new Account"
            />
          </View>,
        ],
        "style": Object {
          "backgroundColor": "#2F95D6",
          "flex": 1,
          "justifyContent": "center",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Image
              source={1}
            />,
            "style": Array [
              Object {
                "flex": 1,
              },
              Object {
                "alignItems": "center",
                "flex": 2,
                "flexDirection": "column",
                "justifyContent": "flex-end",
                "paddingTop": 120,
              },
            ],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "source": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <TextInput
                maxLength={40}
                onChangeText={[Function]}
                placeholder="your@email.com"
                placeholderTextColor="lightgray"
                style={
                  Object {
                    "borderBottomColor": "#2F95D6",
                    "borderBottomWidth": 1,
                    "color": "gray",
                    "height": 50,
                  }
                }
                underlineColorAndroid="#2F95D6"
              />,
              <TextInput
                maxLength={40}
                onChangeText={[Function]}
                placeholder="password"
                placeholderTextColor="lightgray"
                secureTextEntry={true}
                style={
                  Object {
                    "borderBottomColor": "#2F95D6",
                    "borderBottomWidth": 1,
                    "color": "gray",
                    "height": 50,
                  }
                }
                underlineColorAndroid="#2F95D6"
              />,
            ],
            "style": Array [
              Object {
                "flex": 1,
              },
              Object {
                "backgroundColor": "white",
                "flex": 2,
                "flexDirection": "column",
                "paddingBottom": 40,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 32,
              },
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "maxLength": 40,
                "onChangeText": [Function],
                "placeholder": "your@email.com",
                "placeholderTextColor": "lightgray",
                "style": Object {
                  "borderBottomColor": "#2F95D6",
                  "borderBottomWidth": 1,
                  "color": "gray",
                  "height": 50,
                },
                "underlineColorAndroid": "#2F95D6",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "maxLength": 40,
                "onChangeText": [Function],
                "placeholder": "password",
                "placeholderTextColor": "lightgray",
                "secureTextEntry": true,
                "style": Object {
                  "borderBottomColor": "#2F95D6",
                  "borderBottomWidth": 1,
                  "color": "gray",
                  "height": 50,
                },
                "underlineColorAndroid": "#2F95D6",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Button
                backgroundColor="gray"
                disabled={true}
                icon={
                  Object {
                    "name": "sign-in",
                    "type": "font-awesome",
                  }
                }
                onPress={[Function]}
                rounded={true}
                title="Login"
              />,
              <View
                style={
                  Object {
                    "alignItems": "center",
                  }
                }
              >
                <TouchableHighlight
                  activeOpacity={0.85}
                  underlayColor="black"
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    disabled={false}
                    ellipsizeMode="tail"
                  >
                    Forgot password ?
                  </Text>
                </TouchableHighlight>
              </View>,
            ],
            "style": Array [
              Object {
                "flex": 1,
              },
              Object {
                "backgroundColor": "white",
                "flex": 3,
                "flexDirection": "column",
                "justifyContent": "space-around",
                "paddingLeft": 20,
                "paddingRight": 20,
              },
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "backgroundColor": "gray",
                "disabled": true,
                "icon": Object {
                  "name": "sign-in",
                  "type": "font-awesome",
                },
                "onPress": [Function],
                "rounded": true,
                "title": "Login",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <TouchableHighlight
                  activeOpacity={0.85}
                  underlayColor="black"
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    disabled={false}
                    ellipsizeMode="tail"
                  >
                    Forgot password ?
                  </Text>
                </TouchableHighlight>,
                "style": Object {
                  "alignItems": "center",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "activeOpacity": 0.85,
                  "children": <Text
                    accessible={true}
                    allowFontScaling={true}
                    disabled={false}
                    ellipsizeMode="tail"
                  >
                    Forgot password ?
                  </Text>,
                  "underlayColor": "black",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "accessible": true,
                    "allowFontScaling": true,
                    "children": "Forgot password ?",
                    "disabled": false,
                    "ellipsizeMode": "tail",
                  },
                  "ref": null,
                  "rendered": "Forgot password ?",
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Button
              backgroundColor="green"
              icon={
                Object {
                  "name": "user-o",
                  "type": "font-awesome",
                }
              }
              onPress={[Function]}
              rounded={true}
              title="Create new Account"
            />,
            "style": Array [
              Object {
                "flex": 1,
              },
              Object {
                "backgroundColor": "white",
                "flex": 2,
                "justifyContent": "space-around",
                "paddingLeft": 20,
                "paddingRight": 20,
              },
            ],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "backgroundColor": "green",
              "icon": Object {
                "name": "user-o",
                "type": "font-awesome",
              },
              "onPress": [Function],
              "rounded": true,
              "title": "Create new Account",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
